controllers/hardwareclassification_controller.go

```diff
@@ -64,77 +63,140 @@ func (hcReconciler *HardwareClassificationReconciler) Reconcile(req ctrl.Request
-		hwcLog.Error(ErrValidation, ErrValidation.Error())
-		return ctrl.Result{}, nil
-	}
+		// No hosts are using our label, so we can allow the delete to
+		// proceed.
+		hardwareClassification.Finalizers = utils.FilterStringFromList(
+			hardwareClassification.Finalizers, hwcc.Finalizer)
+		err = hcReconciler.Update(context.TODO(), hardwareClassification)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "failed to remove finalizer")
+		}
 
-	//Fetch baremetal host list for the given namespace
-	hostList, bmhList, err := hcManager.FetchBmhHostList(hardwareClassification.ObjectMeta.Namespace)
-	if err != nil {
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusEmpty, hwcc.FetchBMHListFailure, err.Error())
-		hwcLog.Error(err, err.Error())
+		hwcLog.Info("deleting")
 		return ctrl.Result{}, nil
 	}
 
-	if len(hostList) == 0 {
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusEmpty, hwcc.Empty, hwcc.NoBaremetalHost)
-		hwcLog.Info(hwcc.NoBaremetalHost)
-		return ctrl.Result{}, nil
+	// Update our status to report whether we have matched a host or not.
+	status := hwcc.ProfileMatchStatusMatched
+	if matchCount == 0 {
+		status = hwcc.ProfileMatchStatusUnMatched
 	}
-
-	//Extract the hardware details from the baremetal host list
-	validatedHardwareDetails := hcManager.ExtractAndValidateHardwareDetails(extractedProfile, hostList)
-	hwcLog.Info("Validated Hardware Details", "HardwareDetails", validatedHardwareDetails)
-
-	//Compare the host list with extracted profile and fetch the valid host names
-	validHosts := hcManager.MinMaxFilter(hardwareClassification.ObjectMeta.Name, validatedHardwareDetails, extractedProfile)
-	hwcLog.Info("Filtered Bare metal hosts", "ValidHosts", validHosts)
-
-	updateLabelError := hcManager.UpdateLabels(ctx, hardwareClassification.ObjectMeta, validHosts, bmhList)
-
-	if len(updateLabelError) > 0 {
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusEmpty,
-			hwcc.LabelUpdateFailure, strings.Join(updateLabelError, ","))
-		hwcLog.Error(nil, hwcc.UpdateLabelError, "Update Label Error", strings.Join(updateLabelError, ","))
-	} else if len(validHosts) > 0 && len(updateLabelError) == 0 {
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusMatched,
-			hwcc.Empty, hwcc.NOError)
-		hwcLog.Info("Updated profile status", "ProfileMatchStatus", hwcc.ProfileMatchStatusMatched)
-	} else {
-		hcmanager.SetStatus(hardwareClassification, hwcc.ProfileMatchStatusUnMatched,
-			hwcc.Empty, hwcc.NOError)
-		hwcLog.Info("Updated profile status", "ProfileMatchStatus", hwcc.ProfileMatchStatusUnMatched)
+	if hardwareClassification.Status.ProfileMatchStatus != status {
+		hwcLog.Info("updating match status", "newValue", status)
+		hardwareClassification.Status.ProfileMatchStatus = status
+		err = hcReconciler.Status().Update(context.TODO(), hardwareClassification)
+		if err != nil {
+			return ctrl.Result{}, errors.Wrap(err, "failed to update status")
+		}
```

Comment:
With the new implementation there are no longer any errors to report, so we never set the status error fields. It wasn't really clear to me what we were considering an error before, so I just took out the code that was setting the error, but if there is something to set in that field we can replace it.

Reply:
That's fine. Anyway, We are going to implement Error classification so we will cover all the error status related aspect as part of that feature.